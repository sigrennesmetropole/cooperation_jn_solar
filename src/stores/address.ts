import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'

export const useAddressStore = defineStore('address', () => {
  const latitude: Ref<number> = ref(0)
  const longitude: Ref<number> = ref(0)
  const address: Ref<string> = ref('')
  const screenshotAddress: Ref<Blob | null> = ref(null)

  function setLatitudeAndLongitude(
    new_latitude: number,
    new_longitude: number
  ) {
    latitude.value = new_latitude
    longitude.value = new_longitude
  }

  function setAddress(new_address: string) {
    address.value = new_address
  }

  function resetLatitureAndLongitude() {
    setLatitudeAndLongitude(0, 0)
  }

  function resetAddress() {
    setAddress('')
  }

  return {
    latitude,
    longitude,
    setLatitudeAndLongitude,
    address,
    setAddress,
    resetAddress,
    resetLatitureAndLongitude,
    screenshotAddress,
  }
})
