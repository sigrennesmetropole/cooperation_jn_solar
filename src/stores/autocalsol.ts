import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'
import type { AutocalsolResult } from '@/model/autocalsol.model'

export const useAutocalsolStore = defineStore('autocalsolStore', () => {
  const autocalsolResult: Ref<AutocalsolResult | null> = ref(null)

  function setAutocalsolResult(result: AutocalsolResult | null) {
    autocalsolResult.value = result
  }

  function resetAutocalsolResult() {
    autocalsolResult.value = null
  }

  return {
    autocalsolResult,
    setAutocalsolResult,
    resetAutocalsolResult,
  }
})
