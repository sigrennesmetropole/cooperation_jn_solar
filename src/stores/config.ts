import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'
import type { ConfigType } from '@sigrennesmetropole/cooperation_jn_common_ui'

export const useConfigStore = defineStore('configStore', () => {
  const config: Ref<ConfigType | null> = ref(null)

  function setConfig(newConfig: ConfigType | null) {
    config.value = newConfig
  }

  return {
    config,
    setConfig,
  }
})
