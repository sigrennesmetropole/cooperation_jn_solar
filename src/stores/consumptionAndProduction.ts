import type { Ref } from 'vue'
import { ref } from 'vue'
import { defineStore } from 'pinia'

export const useConsumptionAndProductionStore = defineStore(
  'consumptionAndProduction',
  () => {
    const annualConsumption: Ref<number> = ref(0)

    function setAnnualConsumption(new_annual_consumption: number) {
      annualConsumption.value = new_annual_consumption
    }

    function resetAnnualConsumption() {
      setAnnualConsumption(0)
    }

    return {
      annualConsumption,
      setAnnualConsumption,
      resetAnnualConsumption,
    }
  }
)
