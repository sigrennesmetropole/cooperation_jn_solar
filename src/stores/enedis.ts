import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'

export const useEnedisStore = defineStore('enedisStore', () => {
  const isEnedisRedirection: Ref<boolean> = ref(false)

  function setIsEnedisRedirection(newValue: boolean) {
    isEnedisRedirection.value = newValue
  }

  return {
    isEnedisRedirection,
    setIsEnedisRedirection,
  }
})
