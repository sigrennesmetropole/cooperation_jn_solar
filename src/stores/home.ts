import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'

export const useHomeStore = defineStore('home', () => {
  const isTermOfUseAccepted: Ref<boolean> = ref(false)
  const displayError: Ref<boolean> = ref(false)

  function setIsTermOfUseAccepted(newValue: boolean) {
    isTermOfUseAccepted.value = newValue
  }

  function setDisplayError(newValue: boolean) {
    displayError.value = newValue
  }

  return {
    isTermOfUseAccepted,
    displayError,
    setIsTermOfUseAccepted,
    setDisplayError,
  }
})
