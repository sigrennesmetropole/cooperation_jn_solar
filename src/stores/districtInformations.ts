import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'
import type { Viewpoint } from '@vcmap/core'
import type { Feature } from 'ol'
import { useInteractionsStore } from '@/stores/interactions'
import SelectDistrictInteraction from '@/interactions/selectDistrictInteractions'
import { RENNES_LAYER, useLayersStore } from '@/stores/layers'

export const useDistrictStore = defineStore('districtInformations', () => {
  const interactionsStore = useInteractionsStore()
  const layerStore = useLayersStore()

  const districtCode: Ref<number> = ref(0)
  const districtName: Ref<string> = ref('')
  const districtConsumption: Ref<number> = ref(0)
  const districtProduction: Ref<number> = ref(0)
  const districtNumberInstallations: Ref<number> = ref(0)
  const districtPointFeature: Ref<Feature | null> = ref(null)
  const previousViewPoint: Ref<Viewpoint | null> = ref(null)
  const newPointAbscissa: Ref<number> = ref(0)
  const newPointOrdinate: Ref<number> = ref(0)
  const checkboxChecked: Ref<boolean> = ref(false)
  const canBeDisplayed: Ref<boolean> = ref(true)
  function setDistrictIrisCode(newDistrictCode: number) {
    districtCode.value = newDistrictCode
  }

  function setDistrictName(newDistrictName: string) {
    districtName.value = newDistrictName
  }

  function setDistrictConsumption(newDistrictConsumption: number) {
    districtConsumption.value = newDistrictConsumption
  }

  function setDistrictProduction(newDistrictProduction: number) {
    districtProduction.value = newDistrictProduction
  }

  function setDistrictNumberInstallations(
    newDistrictNumberInstallations: number
  ) {
    districtNumberInstallations.value = newDistrictNumberInstallations
  }

  function setDistrictInformations(
    newDistrictCode: number,
    newDistrictName: string,
    newDistrictConsumption: number,
    newDistrictProduction: number,
    newDistrictNumberInstallations: number
  ) {
    setDistrictIrisCode(newDistrictCode)
    setDistrictName(newDistrictName)
    setDistrictConsumption(newDistrictConsumption)
    setDistrictProduction(newDistrictProduction)
    setDistrictNumberInstallations(newDistrictNumberInstallations)
  }

  function setNewPointFeatureOnSelectedDistrict(newPointFeature: Feature) {
    districtPointFeature.value = newPointFeature
  }

  function setNewCoordinates(newAbscissa: number, newOrdinate: number) {
    newPointAbscissa.value = newAbscissa
    newPointOrdinate.value = newOrdinate
  }

  function setCheckboxChecked(newAction: boolean) {
    checkboxChecked.value = newAction
    if (newAction && canBeDisplayed) {
      layerStore.enableLayer(RENNES_LAYER.iris)
      interactionsStore.enableInteraction(SelectDistrictInteraction)
    } else {
      layerStore.disableLayer(RENNES_LAYER.iris)
      interactionsStore.disableInteraction(SelectDistrictInteraction)
    }
  }

  function setCanBeDisplayed(isFarEnough: boolean) {
    canBeDisplayed.value = isFarEnough
    if (!isFarEnough) {
      setCheckboxChecked(false)
    }
  }

  function resetDistrictStore() {
    setDistrictInformations(0, '', 0, 0, 0)
  }

  return {
    districtCode,
    districtName,
    districtConsumption,
    districtProduction,
    districtNumberInstallations,
    districtPointFeature,
    previousViewPoint,
    newPointAbscissa,
    newPointOrdinate,
    checkboxChecked,
    canBeDisplayed,
    setDistrictIrisCode,
    setDistrictName,
    setDistrictConsumption,
    setDistrictProduction,
    setDistrictNumberInstallations,
    setDistrictInformations,
    setNewPointFeatureOnSelectedDistrict,
    setNewCoordinates,
    setCheckboxChecked,
    setCanBeDisplayed,
    resetDistrictStore,
  }
})
