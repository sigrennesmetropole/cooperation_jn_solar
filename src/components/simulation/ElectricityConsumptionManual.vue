<script setup lang="ts">
import { ref } from 'vue'
import { useConsumptionAndProductionStore } from '@/stores/consumptionAndProduction'

const consumption = ref('')
const consumptionAndProductionStore = useConsumptionAndProductionStore()

const storeConsumption = () => {
  consumptionAndProductionStore.setAnnualConsumption(
    parseInt(consumption.value)
  )
}

function inputIsCompleted() {
  const inputLength = consumption.value.length
  if (inputLength <= 0) {
    return false
  } else {
    return true
  }
}
</script>

<template>
  <div class="flex flex-col gap-1.5 mt-10">
    <p class="font-dm-sans font-bold text-lg">
      Ma consommation annuelle est de :
    </p>
    <div class="flex flex-row">
      <input
        class="border border-neutral-400 rounded h-[42px] px-3 py-[9px] text-black font-dm-sans font-medium placeholder:text-neutral-600 placeholder:text-base placeholder:font-dm-sans w-full"
        type="number"
        placeholder="kWh"
        v-model="consumption"
        @keyup="storeConsumption"
      />
      <div
        v-if="inputIsCompleted()"
        append-unit="kWh"
        class="text-base font-dm-sans font-medium relative after:content-[attr(append-unit)] after:absolute after:right-2 after:top-2"
      ></div>
    </div>
    <div class="mt-10">
      <div
        class="relative flex flex-col mx-auto h-[200px] pr-1 overflow-auto scrollbar-thin scrollbar-w-[3px] scrollbar-thumb-red-600 scrollbar-track-neutral-400"
      >
        <p class="font-dm-sans font-bold text-neutral-900 text-sm">
          Protection des données personnelles :
        </p>
        <p class="font-dm-sans font-medium text-neutral-600 text-xs">
          Nous sollicitons via ce formulaire des données personnelles vous
          concernant afin de simuler l'installation de panneaux photovoltaïque
          et calculer vos économies d'énergie.Ces données ne sont pas conservées
          au-delà du calcul effectué lors de l'utilisation du site. Les données
          sont par la suite soit supprimées immédiatement ou anonymisées à des
          fins statistiques. La base légale du traitement est le consentement de
          l'usager. Ces données sont transmises pour la durée de la simulation à
          l'API Autocalsol de l'INES.Conformément au Règlement général sur la
          protection des données (RGPD) (UE) n°2016/679 du 27 avril 2016 et à la
          loi n°2018-493 du 20 juin 2018 relative à la protection des données
          personnelles, vous pouvez accéder aux données vous concernant où
          demander leur effacement. Vous disposez également d'un droit
          d’opposition, d’un droit de rectification, d’un droit à la limitation
          du traitement de vos données ainsi que d'un droit au retrait de votre
          consentement. Pour exercer ces droits ou pour toute question sur le
          traitement de vos données, vous pouvez contacter le Délégué à la
          protection des données de Rennes (Ville et Métropole) via le
          formulaire de contact "e-démarches" du site internet
          <a
            class="underline decoration-1 font-medium"
            href="https://metropole.rennes.fr"
            target="_blank"
            >(https://metropole.rennes.fr)</a
          >
          ou par voie postale à adresser à : Madame le Présidente de Rennes
          Métropole - A l'attention du Délégué à la protection des données - 4
          avenue Henri Fréville -CS93111 - 35031 Rennes Cedex Plus
          d’informations sur :
          <a
            class="underline decoration-1 font-medium"
            href="https://www.cnil.fr/"
            target="_blank"
            >https://www.cnil.fr</a
          >
        </p>
      </div>
    </div>
  </div>
</template>

<style scoped>
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
}
input[type='number'] {
  -moz-appearance: textfield;
}
</style>
