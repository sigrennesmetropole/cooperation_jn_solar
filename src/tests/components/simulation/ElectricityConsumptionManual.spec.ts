import ElectricityConsumptionManual from '@/components/simulation/ElectricityConsumptionManual.vue'
import { describe, it, expect, beforeEach, vi } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'
import { useConsumptionAndProductionStore } from '@/stores/consumptionAndProduction'

describe('ElectricityConsumptionManual.vue', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(ElectricityConsumptionManual, {
      global: {
        plugins: [
          createTestingPinia({
            createSpy: vi.fn,
            stubActions: false,
            stubPatch: false,
            fakeApp: true,
          }),
        ],
      },
    })
  })

  it('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('stores consumption value when input changes', async () => {
    const consumptionInput = wrapper.find('input[type="number"]')
    await consumptionInput.setValue('1500')
    await consumptionInput.trigger('keyup')
    const consumptionAndProductionStore = useConsumptionAndProductionStore()
    expect(consumptionAndProductionStore.annualConsumption).toBe(1500)
  })

  it('shows kWh unit when input is completed', async () => {
    const consumptionInput = wrapper.find('input[type="number"]')
    await consumptionInput.setValue('1500')
    const unitElement = wrapper.find('[append-unit="kWh"]')
    expect(unitElement.exists()).toBe(true)
  })

  it('hides kWh unit when input is empty', async () => {
    const consumptionInput = wrapper.find('input[type="number"]')
    await consumptionInput.setValue('')
    const unitElement = wrapper.find('[append-unit="kWh"]')
    expect(unitElement.exists()).toBe(false)
  })
})
