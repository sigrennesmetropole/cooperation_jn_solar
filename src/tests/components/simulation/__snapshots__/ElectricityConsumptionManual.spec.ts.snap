// Vitest Snapshot v1

exports[`ElectricityConsumptionManual.vue > renders the component correctly 1`] = `
"<div data-v-1452b0f4=\\"\\" class=\\"flex flex-col gap-1.5 mt-10\\">
  <p data-v-1452b0f4=\\"\\" class=\\"font-dm-sans text-sm\\">Votre consommation annuelle</p>
  <div data-v-1452b0f4=\\"\\" class=\\"flex flex-row\\"><input data-v-1452b0f4=\\"\\" class=\\"border border-neutral-400 rounded h-[42px] px-3 py-[9px] text-black font-dm-sans text-medium placeholder:text-neutral-600 placeholder:text-base placeholder:font-dm-sans w-full\\" type=\\"number\\" placeholder=\\"kWh\\">
    <!--v-if-->
  </div>
</div>"
`;
